docs/docs/segment-haskell.md

```diff
@@ -0,0 +1,61 @@
+---
+id: haskell
+title: Haskell
+sidebar_label: Haskell
+---
+
+## What
+
+Display the currently active Glasgow Haskell Compiler (GHC) version.
+
+## Sample Configuration
+
+```json
+{
+  "type": "haskell",
+  "style": "powerline",
+  "powerline_symbol": "\uE0B0",
+  "foreground": "#906cff",
+  "background": "#100e23",
+  "properties": {
+    "template": " \ue61f {{ .Full }}"
+  }
+}
+```
+
+## Properties
+
+- home_enabled: `boolean` - display the segment in the HOME folder or not - defaults to `false`
+- fetch_version: `boolean` - display the GHC version - defaults to `true`
+- display_error: `boolean` - show the error context when failing to retrieve the version information - defaults to `true`
+- missing_command_text: `string` - text to display when the command is missing - defaults to empty
+- display_mode: `string` - determines when the segment is displayed
+  - `always`: the segment is always displayed
+  - `files`: the segment is only displayed when `*.hs`, `*.lhs`, `stack.yaml`, `package.yaml`, `*.cabal`, or `cabal.project` files are present (default)
+- use_stack_ghc: `string` - determines when to use stack ghc to retrieve the version information. Using stack ghc will decrease performance.
```

Comment:
can we rename this to `stack_ghc_mode`? That way it's in line with `display_mode`.

Reply:
changed everything related to `use_stack_ghc` to fit with `stack_ghc_mode`